src/ru/nsu/ccfit/zuev/skins/SkinData.java

```diff
@@ -0,0 +1,40 @@
+package ru.nsu.ccfit.zuev.skins;
+
+
+import androidx.annotation.NonNull;
+
+import org.json.JSONObject;
+
+import ru.nsu.ccfit.zuev.osu.datatypes.DefaultData;
+import ru.nsu.ccfit.zuev.osu.datatypes.IDefaultableData;
+
+public abstract class SkinData<I> implements IDefaultableData<I> {
+    protected DefaultData<I> data;
+    public String tag;
```

Comment:
Is this not intended to be `public`? It's weird to have a `getTag()` method that returns this value when you can simply do `<SkinData<I>>.tag`.

Reply:
Oh yes.